tests/test_deployments.py

```diff
@@ -30,6 +30,19 @@ async def test_that_infrastructure_defaults_to_process(self):
         d = Deployment(name="foo")
         assert isinstance(d.infrastructure, Process)
 
+    async def test_infrastructure_accepts_arbitrary_infra_types(self):
+        class CustomInfra(Infrastructure):
+            type = "CustomInfra"
+
+            def run(self):
+                return 42
+
+            def preview(self):
+                return "woof!"
+
+        d = Deployment(name="foo", infrastructure=CustomInfra())
```

Comment:
Can you add a test with `infrastructure=CustomInfra().dict()` too? We need that to ensure it works from YAML.

Reply:
(or an actual test that loads from a YAML config which would be better)

Reply:
@peytonrunyan want to make sure you've seen this as we kick off the new week